### ISSUE TITLE
[A11y] Advanced Search "select" missing label

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1789]
### Operating System

Windows

### Browser

Edge

### Affected page

Advanced Search

### Admin user?

- [X] Yes

### Issue Description

The "Find items that match [all / any] of:" bit isn't accessible.  The dropdown has no label, so a SR user has to find surrounding context.

### Suggested Fix

Mid-sentence dropdowns are notoriously tricky to do correctly, especially when the select is actually defining scope for other fields.  It is probably not horrible to have a hidden label like "Search scope", but it might still be a pretty bad experience, and may not even conform to WCAG - this is a fuzzy issue.

More conformant but potentially a lot more work would be to put the field below the others, make it a radio group, have a fieldset with a legend like "Search scope", and the options "All of the above fields must match" versus "Any of the above fields must match".

(Note that the terms "above" and "below" *are accessible* despite implying a location, because they are so often used to mean "before" and "after" in textual contexts)

**Comment by CGillen** [comment]
@jechols Would changing the sentence text to "Find items that match `all/any below fields`" work, with the label `Search scope`? So the context looks like Search scope: All below fields

**Comment by jechols** [comment]
Of some relevance: https://webaim.org/discussion/mail_thread?thread=8693#post4 and https://webaim.org/discussion/mail_message?id=39696

Unfortunately if you dig through threads like that, nobody ever comes to a consensus.  "Fill-in-the-blank" fields are just notoriously tough to do right.

But Birkir seems to know a TON about accessibility, so I'm thinking we could go with a modification of his suggestion (at the time he hadn't tried it out).  Here's what I came up with, which works ... moderately well with NVDA:

```html
<h3>
  <label class="query-criteria-heading">
    Find items that match
    <select name="op" id="op" class="input-small" aria-description="scope">
      <option value="AND">all</option>
      <option value="OR">any</option>
    </select> of 
  </label>
</h3>
```

Using description instead of label ensures the label is still the full text while giving that tiny bit of added context.  And I think this situation is still far from ideal (and possibly still very difficult to use unfortunately), but it may be the best option we have until there are better patterns for this kind of thing....

**Comment by CGillen** [comment]
Second fix: #1911
